src/main/java/hlf/java/rest/client/service/impl/EventPublishServiceImpl.java

```diff
@@ -86,9 +91,17 @@ public boolean publishChaincodeEvents(
     boolean status = true;
 
     try {
-
+        String key = String.valueOf(payload.hashCode());
+        if (fabricProperties.getEvents().isStandardCCEventEnabled()) {
+            // Fetch the key information for chaincode events,
+            // but only if the feature is enabled.
+
+            // Parse the payload and use the key.
+            StandardCCEvent standardCCEvent = FabricEventParseUtil.parseString(payload, StandardCCEvent.class);
+            key = standardCCEvent.getKey();
+        }
       ProducerRecord<String, String> producerRecord =
-          new ProducerRecord<>(topicName, String.valueOf(payload.hashCode()), payload);
+          new ProducerRecord<>(kafkaProperties.getEventListener().getTopic(), key, payload);
```

Comment:
Chaincode developers would utilize the SDK released by the connector to structure events in a standard format. The reason for leaving the payload part as is, is to be transparent in the connector layer for passing on the event information generated in the chaincode.

Reply:
Okay, I assumed that the `event` property within `StandardCCEvent` is what the subscriber would receive. I believe this would also force the downstream subscribers to be aware about `StandardCCEvent` for deserialising the and extracting the Event.